app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/labtest/LabTestDetailFragment.kt

```diff
@@ -44,67 +45,99 @@ class LabTestDetailFragment : BaseRecordDetailFragment(R.layout.fragment_lab_tes
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        setupComposeToolbar(binding.composeToolbar.root)
-        setUpRecyclerView(args.hdid)
-        viewModel.getLabTestDetails(args.labOrderId)
-        observeUiState()
         observePdfData()
-        initComments()
+        observeCommentsSyncCompletion()
     }
 
-    override fun getCommentView(): AddCommentLayout = binding.comment
+    override fun onResume() {
+        super.onResume()
+        viewModel.getLabTestDetails(args.labOrderId)
+    }
 
-    override fun getScrollableView() = binding.rvLabTestDetailList
+    @Composable
+    override fun GetComposableLayout() {
 
-    override fun getCommentEntryTypeCode() = CommentEntryTypeCode.LAB_RESULTS
+        val uiState = viewModel.uiState.collectAsState().value
 
-    override fun getParentEntryId(): String? = viewModel.uiState.value.parentEntryId
+        var commentState: CommentsUiState? = null
 
-    override fun getProgressBar(): View = binding.progressBar
+        if (BuildConfig.FLAG_ADD_COMMENTS) {
+            uiState.parentEntryId?.let { commentsViewModel.getComments(it) }
+            commentState = commentsViewModel.uiState.collectAsState().value
+        }
 
-    private fun setUpRecyclerView(hdid: String?) {
-        labTestDetailAdapter = LabTestDetailAdapter(
-            viewPdfClickListener = { viewModel.getLabTestPdf(hdid) }
-        )
+        MyHealthScaffold(
+            title = uiState.toolbarTitle,
+            isLoading = uiState.onLoading,
+            navigationAction = { findNavController().popBackStack() }
+        ) {
+            LabTestScreen(
+                uiState = uiState,
```

Comment:
Should pass viewModel inside LabTestScreen instead of UiState.

Reply:
done